rust-src/concordium_base/src/id/sigma_protocols/vcom_eq.rs

```diff
@@ -0,0 +1,413 @@
+//! The module provides the implementation of the `vcom_eq` sigma
+//! protocol. This protocol enables one to prove knowledge of
+//! $x_1, ..., x_n, r, (r_i)_{i\in I}$ such that $ C = h^r \prod_{i=1}^n
+//! g_i^{x_i}$ and $C_i = \bar{g}^{x_i} \bar{h}^{r_i}$ for $i\in I$.
+use super::common::*;
+use crate::{
+    common::*,
+    curve_arithmetic::{multiexp, Curve},
+    random_oracle::{Challenge, RandomOracle},
+};
+use ff::Field;
+use itertools::izip;
+use std::{collections::BTreeMap, rc::Rc};
+
+type IndexType = u8;
+
+/// The public information known to both the prover and the verfier.
+pub struct VecComEq<C: Curve> {
```

Comment:
The implementation requires C and C_i-s to be on the same curve. This is of course fine for the first version of our verifiable credentials, but I think in future, it would be a benefit if we can have vcom_eq with commitments on possibly different curves.

Reply:
We discussed this briefly with @eb-concordium last week, and since we have no other curves we will keep it simple for now.

The computations don't become more complex of course, but all type signatures, etc, just become extra bloated.

Reply:
Ok, I see. Makes sense.

Reply:
Yes. We can always add it if needed.